contracts/EntryPoint.sol

```diff
@@ -139,7 +132,7 @@ contract EntryPoint is StakeManager {
     struct UserOpInfo {
         bytes32 requestId;
         uint256 prefund;
-        PaymentMode paymentMode;
+        address paymaster;
```

Comment:
Also, I don't think there is a single function here that has `UserOpInfo` but does not have a corresponding `UserOperation`, making this field a duplicate.

Reply:
it isn't. its an in-memory collected data about a userOp (context, prefund, etc)
it also caches "paymaster" address.